### PR TITLE
Search block

### PR DESCRIPTION
**Pull request by spolu**
cc @mauriyouth @aickin

Based off and supersedes #23 #24

### CODE REVIEW

**Comment by spolu** [comment]
Also added you both as collaborators on the repo so that you can clone it and PR to it direclty if you wish

File: core/src/blocks/search.rs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                Rule::expected => {
+                    Err(anyhow!("`expected` is not yet supported in `search` block"))?
+                }
+                _ => unreachable!(),
+            }
+        }
+
+        if !query.is_some() {
+            Err(anyhow!(
+                "Missing required `question` in `googleanswer` block"
```
Comment (mauriyouth) [inline core/src/blocks/search.rs:48]: missing `query`for `search block, same for `engine`

File: front/lib/specification.js [hunk 2]
```diff
@@ -246,6 +259,15 @@ export function dumpSpecification(spec, latestDatasets) {
         out += "\n";
         break;
       }
+
+      case "search": {
+        out += `search ${block.name} {\n`;
+        out += `query: \n\`\`\`\n${block.spec.query}\n\`\`\`\n`;
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/lib/specification.js:263]: For now I rely on the backend default but will add later yes! :+1:

**Comment by spolu** [comment]
@aickin server-side checking looks great \o/

**Pull request merged**